solidity/test/Oracle_test.js

```diff
@@ -146,43 +146,111 @@ contract('Oracle', () => {
     let mock, internalId
     let requestId = 'XID'
 
-    beforeEach(async () => {
-      mock = await deploy('examples/GetterSetter.sol')
-      let fHash = functionSelector('requestedBytes32(bytes32,bytes32)')
-      let args = requestDataBytes(specId, mock.address, fHash, requestId, '')
-      let req = await requestDataFrom(oc, link, 0, args)
-      internalId = req.receipt.logs[2].topics[1]
-    })
+    context('successful consumer', () => {
+      beforeEach(async () => {
+        mock = await deploy('examples/GetterSetter.sol')
+        let fHash = functionSelector('requestedBytes32(bytes32,bytes32)')
+        let args = requestDataBytes(specId, mock.address, fHash, requestId, '')
+        let req = await requestDataFrom(oc, link, 0, args)
```

Comment:
So this still bypasses the regular `requestData` call to manually stitch up the fulfillment scenario. It also uses `GetterSetter` rather than malicious contract. Like below

```
const req = await mock.requestData('assertFail(bytes32,bytes32)')
```

This all just means that we don't have a test that does the happy path of request -> fulfill, without bypassing a code path with manual setup. I'm actually ok with that since we explicitly cover request then fulfill, but it's worth bringing up.

Reply:
That contract doesn't have the requestData function. We could add it, but it seems kind of outside of the scope of this story.